pytorch_lightning/compressor/intel/neural_compressor/quantization.py

```diff
@@ -0,0 +1,119 @@
+from enum import Enum
+from torch.utils.data import DataLoader
+from pytorch_lightning.core.lightning import LightningModule
+from typing import Optional
+import pytorch_lightning as pl
+
+
+class QuantizationMode(Enum):
+    PTQ_DYNAMIC = "post_training_dynamic_quant"
+    PTQ_STATIC = "post_training_static_quant"
+    QAT = "quant_aware_training"
+
+
+class IncQuantization:
```

Comment:
This should be refactored into a callback. We won't merge objects wrapping a Trainer. The Trainer should be the entry point for anything else.

Reply:
done